### PR TITLE
[DROOLS-6907] Arithmetic operation with BigDecimal in constraint fail…

### PR DESCRIPTION
**Pull request by tkobayas**
…s in executable-model

**Ports** 
This PR is for 7.x
for main -> https://github.com/kiegroup/drools/pull/4325

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6907

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/ConstraintParser.java [hunk 1]
```diff
@@ -814,13 +814,13 @@ private static boolean isAnyOperandNullLiteral(TypedExpression left, TypedExpres
...
-        compareMethod.addArgument( toBigDecimalExpression( left ) );
-        compareMethod.addArgument( toBigDecimalExpression( right ) );
+        compareMethod.addArgument( toBigDecimalExpression( left, context ) );
+        compareMethod.addArgument( toBigDecimalExpression( right, context ) );
         return new SpecialComparisonResult(compareMethod, left, right);
     }
 
     // TODO luca this logic should be moved in Constraint compiler?
-    private Expression toBigDecimalExpression( TypedExpression typedExpression) {
+    public static Expression toBigDecimalExpression( TypedExpression typedExpression, RuleContext context) {
```
Comment (mariofusco) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/ConstraintParser.java:823]: The newly added RuleContext arg seems not used anywhere, please remove it.

**Review by mariofusco — APPROVED** [review]
I just added one minor comment, for the rest it looks ok, thanks.

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/ConstraintParser.java (hunk 1 above)
Comment (tkobayas) [reply drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/ConstraintParser.java:823]: @mariofusco  `RuleContext context` is used in the method. Originally, the method referred to a private field `ConstraintParser.context` but by changing the method to public static, I have to pass it as an argument.
```
    public static Expression toBigDecimalExpression( TypedExpression typedExpression, RuleContext context) {
        MethodCallExpr toBigDecimalMethod = new MethodCallExpr( null, "org.drools.modelcompiler.util.EvaluationUtil.toBigDecimal" );
        Expression arg = typedExpression.getExpression();

        Optional<Class<?>> originalPatternType = typedExpression.getOriginalPatternType();

        ConstraintCompiler constraintCompiler = createConstraintCompiler(context, originalPatternType); // <- Here
```

Comment (mariofusco) [reply drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/ConstraintParser.java:823]: Sorry, I didn't realize this.

**Pull request merged**